AND-SOPT-iOS/AND-SOPT-iOS/Assignment/Week02-03/FinanceView/Cell/TopPaidFreeAppCell.swift

```diff
@@ -0,0 +1,130 @@
+        let label = UILabel()
+        label.font = .systemFont(ofSize: 14)
+        label.textColor = .lightGray
+        label.numberOfLines = 2
+        return label
+    }()
+    
+    private let downloadStateButton: UIButton = {
+        let button = UIButton()
+        button.setTitleColor(.systemBlue, for: .normal)
+        button.backgroundColor = .tertiarySystemGroupedBackground
+        button.layer.cornerRadius = 18
+        button.clipsToBounds = true
+        button.tintColor = .systemBlue
+        button.titleLabel?.font = UIFont.systemFont(ofSize: 15, weight: .heavy)
+        return button
+    }()
+    
+    private let inAppPurchasesLabel: UILabel = {
+        let label = UILabel()
+        label.text = "앱 내 구입"
+        label.font = .systemFont(ofSize: 10)
+        label.textColor = .lightGray
+        label.isHidden = true
+        return label
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setUI()
+        setLayout()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func setUI() {
+        addSubviews(logoImageView, rankLabel, titleLabel, descriptionLabel, downloadStateButton, inAppPurchasesLabel)
+    }
+    
+    private func setLayout() {
+        logoImageView.snp.makeConstraints {
+            $0.top.equalToSuperview()
+            $0.leading.equalToSuperview().offset(15)
+            $0.width.height.equalTo(65)
+        }
+        
+        rankLabel.snp.makeConstraints {
+            $0.top.equalToSuperview().offset(8)
+            $0.leading.equalTo(logoImageView.snp.trailing).offset(10)
+        }
+        
+        titleLabel.snp.makeConstraints {
+            $0.top.equalToSuperview().offset(8)
+            $0.leading.equalTo(rankLabel.snp.trailing).offset(10)
+        }
+        
+        descriptionLabel.snp.makeConstraints {
+            $0.top.equalTo(titleLabel.snp.bottom).offset(5)
+            $0.leading.equalTo(titleLabel)
+        }
+        
+        downloadStateButton.snp.makeConstraints{
+            $0.trailing.equalToSuperview().offset(-20)
+            $0.centerY.equalTo(logoImageView)
+            $0.width.equalTo(86)
+            $0.height.equalTo(35)
+        }
+        
+        inAppPurchasesLabel.snp.makeConstraints{
+            $0.top.equalTo(downloadStateButton.snp.bottom).offset(5)
+            $0.centerX.equalTo(downloadStateButton)
+        }
+    }
+    
+    func bind(_ financeAppList: FinanceAppList) {
+        logoImageView.image = financeAppList.logoImage
+        titleLabel.text = financeAppList.title
+        descriptionLabel.text = financeAppList.description
+        downloadStateButton.setTitle(financeAppList.downloadState, for: .normal)
+        inAppPurchasesLabel.isHidden = !financeAppList.isInAppPurchases
+        rankLabel.isHidden = !financeAppList.isRank
+        
+        if let rank = financeAppList.rank {
```

Comment:
```suggestion
        if let rank = rank {
        rankLabel.text = "\(rank)"
        rankLabel.isHidden = false
    } else {
        rankLabel.isHidden = true
    }
```
text에 nil 값을 할당하는 것보단 isHidden으로 설정하는 게 좀 더 직관적일 것 같다고 생각합니다.

Reply:
참고하겠습니다!